File: server/src/main/java/com/hospitaltender/server/service/TenderService.java
package com.hospitaltender.server.service;

import com.hospitaltender.server.dto.request.CreateTenderRequest;
import com.hospitaltender.server.dto.request.UpdateTenderRequest;
import com.hospitaltender.server.dto.response.TenderResponse;
import com.hospitaltender.server.enums.TenderStatus;

import java.util.List;

public interface TenderService {

    TenderResponse create(CreateTenderRequest request);

    TenderResponse getById(Long id);

    List<TenderResponse> getAll();

    List<TenderResponse> getByStatus(TenderStatus status);

    TenderResponse update(Long id, UpdateTenderRequest request);

    TenderResponse updateStatus(Long id, TenderStatus status);

    void delete(Long id);
}

File: ui/hospital-system-ui/src/services/tender.service.ts
import api from './api'
import { TenderResponse, CreateTenderRequest, UpdateTenderRequest, TenderStatus } from '@/types'

export const tenderService = {
  getAll: () =>
    api.get<TenderResponse[]>('/tenders').then((res) => res.data),

  getById: (id: number) =>
    api.get<TenderResponse>(`/tenders/${id}`).then((res) => res.data),

  getByStatus: (status: TenderStatus) =>
    api.get<TenderResponse[]>(`/tenders/status/${status}`).then((res) => res.data),

  create: (data: CreateTenderRequest) =>
    api.post<TenderResponse>('/tenders', data).then((res) => res.data),

  update: (id: number, data: UpdateTenderRequest) =>
    api.put<TenderResponse>(`/tenders/${id}`, data).then((res) => res.data),

  updateStatus: (id: number, status: TenderStatus) =>
    api.patch<TenderResponse>(`/tenders/${id}/status`, null, { params: { status } }).then((res) => res.data),

  delete: (id: number) =>
    api.delete(`/tenders/${id}`),
}
